cockatrice/src/player.cpp

```diff
@@ -1133,43 +1133,16 @@ void Player::actCreateRelatedCard()
     relatedCards.append(sourceCard->getInfo()->getReverseRelatedCards2Me());
     CardRelation * cardRelation = relatedCards.at(action->data().toInt());
 
-    const QString &displayName = cardRelation->getName();
-    QString dbName = dbNameFromTokenDisplayName(displayName);
-    if(cardRelation->getIsVariable()) {
-        bool ok;
-        dialogSemaphore = true;
-        int count = QInputDialog::getInt(0, tr("Create tokens"), tr("Number:"), cardRelation->getDefaultCount(), 1, MAX_TOKENS_PER_DIALOG, 1, &ok);
-        dialogSemaphore = false;
-        if (!ok)
-            return;
-        for (int i = 0; i < count; i++) {
-            createCard(sourceCard, dbName);
-        }
-    } else if(cardRelation->getDefaultCount() > 1) {
-        for (int i = 0; i < cardRelation->getDefaultCount(); i++) {
-            createCard(sourceCard, dbName);
-        }
-    } else {
-        if (cardRelation->getDoesAttach()) {
-            createAttachedCard(sourceCard, dbName);
-        } else {
-            createCard(sourceCard, dbName);
-        }
-    }
-
    /*
-    * If we made a token via "Token: TokenName"
-    * then lets allow it to be created via create another
+    * If we make a token via "Token: TokenName"
+    * then let's allow it to be created via "create another token"
     */
-    if (cardRelation->getCanCreateAnother()) {
-        CardInfo *cardInfo = db->getCard(dbNameFromTokenDisplayName(displayName));
+    if (createRelatedFromRelation(sourceCard, cardRelation) && cardRelation->getCanCreateAnother()) {
+        CardInfo *cardInfo = db->getCard(dbNameFromTokenDisplayName(cardRelation->getName()));
         setLastToken(cardInfo);
```

Comment:
New method implements null check and can't return true unless both arguments aren't null.